feat: Add native Git hooks for automatic code formatting

Hell1213 opened the pull request:
### Description

This PR implements native Git pre-commit hooks to automatically format code before committing, with automated setup via npm postinstall script.

**Changes made:**
=> Created `.githooks/pre-commit` for native Git hook implementation
=> Added `postinstall` script in `frontend/package.json` to automatically configure Git hooks on `npm install`
=> Pre-commit hook automatically runs:
  - Prettier on staged frontend files (JS/TS/JSON/CSS/MD) via `npm run lint`
  - gofmt on staged backend Go files (if Go is installed)
=> Updated `CONTRIBUTING.md` with automated setup instructions
=> Updated `development/README.md` to mention automatic Git hooks setup
=> Removed `scripts/setup-git-hooks.sh` as hooks now configure automatically

**How it works:**
When contributors run `npm install` in the frontend directory, the `postinstall` script automatically configures Git to use the `.githooks` directory. The pre-commit hook then:
1. Detects staged files
2. Runs `npm run pre-commit` → `npm run lint` → `npx prettier --write .` for frontend files
3. Runs `gofmt` for backend Go files
4. Re-stages the formatted files

This ensures consistent code style without any manual setup or intervention.

**Why this approach:**
Based on maintainer feedback, this implementation:
=> Uses native Git hooks instead of Husky (avoids root package.json complexity)
=> Automatically configures on `npm install` (no manual setup script needed)
=> Follows the exact pattern suggested by the maintainer

**Testing:**
-  Tested with badly formatted frontend files - automatically formatted on commit
-  Verified hook only formats staged files
- Confirmed hook executes before commit and re-stages formatted files
-  Gracefully handles missing gofmt (shows warning, doesn't fail)
-  Frontend tests pass (126 tests)

- Fixes: #114

### Checklist

- [x] Ran `npm run lint` (for formatting)
- [x] Ran `gofmt -w .` (for Go backend) - *Note: Go not installed on test system, but hook is configured correctly*
- [x] Ran `npm test` (for JS/TS testing) - All 126 tests passed
- [ ] Added unit tests, if applicable - *Not applicable for Git hooks setup*
- [x] Verified all tests pass
- [x] Updated documentation, if needed

### Additional Notes

**For Contributors:**
After pulling this PR, simply run `npm install` in the frontend directory. The Git hooks will be automatically configured, and your code will be formatted on every commit.

**For Reviewers:**
You can test the hook by:
1. Running `npm install` in the frontend directory to configure the hook
2. Creating a badly formatted file in `frontend/src/`
3. Staging it with `git add`
4. Committing and watching it format automatically

This implementation matches the maintainer's requested approach exactly, using native Git hooks with automated setup via npm postinstall.

Hell1213 commented:
hey @its-me-abhishek .

This PR is ready for review. I've implemented Husky pre-commit hooks to automatically format code before committing, which solves issue #114.

its-me-abhishek commented:
hi, while this looks fine, i think having another package.json will just make this project more complicated. i think we can shall either use a Githook instead of Husky, or if we continue to integrate this Husky into the package.json of frontend dir (which will not work with the root dir so thats a drawback).

Hell1213 commented:
hey @its-me-abhishek ,

thanks for the feedback! You're absolutely right - adding a root package.json does add unnecessary complexity. 

I'll refactor this to use native Git hooks instead of Husky. Will update the PR soon.

Hell1213 commented:
> hi, while this looks fine, i think having another package.json will just make this project more complicated. i think we can shall either use a Githook instead of Husky, or if we continue to integrate this Husky into the package.json of frontend dir (which will not work with the root dir so thats a drawback).

hey @its-me-abhishek,
I've updated the PR based on your feedback. Replaced Husky with native Git hooks to avoid the root package.json complexity.

its-me-abhishek commented:
This looks fine but  will add another caveat or layer, to setup inorder to make things work.

It'd be better to add something similar this to the githooks directory along with go fmt

```
#! /bin/sh 

echo "This runs before the commit to test frontend linting!"
cd frontend
npm run pre-commit
```

and then add these scripts to the frontends' package.json

```
    "postinstall": "git config core.hooksPath ../.githooks",
    "pre-commit": "npm run lint",
```

It'll be setup automatically, then, after the user runs npm install, hence making it an automated workflow!
